curvefs/src/volume/block_device_aio.h

```diff
@@ -0,0 +1,144 @@
+/*
+ *  Copyright (c) 2022 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Date: Thursday Jul 07 14:56:25 CST 2022
+ * Author: wuhanqing
+ */
+
+#ifndef CURVEFS_SRC_VOLUME_BLOCK_DEVICE_AIO_H_
+#define CURVEFS_SRC_VOLUME_BLOCK_DEVICE_AIO_H_
+
+#include <atomic>
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+
+#include "include/client/libcurve_define.h"
+
+namespace curve {
+namespace client {
+class FileClient;
+}  // namespace client
+}  // namespace curve
+
+namespace curvefs {
+namespace volume {
+
+using ::curve::client::FileClient;
+
+struct AioRead {
+    // aio context
+    CurveAioContext aio;
+
+    // original request context
+    off_t offset;
+    size_t length;
+    char* data;
+
+    // backend block device
+    FileClient* dev;
+    int fd;
+
+    // synchronization
+    bool done = false;
+    std::mutex mtx;
+    std::condition_variable cond;
+
+    struct Padding {
+        off_t offset;
+        size_t length;
+        std::unique_ptr<char[]> data;
+    };
+
+    // padding read request if necessary
+    std::unique_ptr<Padding> padding;
```

Comment:
need init nullptr when init？

Reply:
it's nullptr by default, the default constructor guarantee this.